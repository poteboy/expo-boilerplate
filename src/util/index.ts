export * from './test';
export * from './validation';
