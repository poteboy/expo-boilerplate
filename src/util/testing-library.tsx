import React, { FC } from 'react';
import { NativeBaseProvider } from 'native-base';
import { theme } from '@src/styles';
import {
  render as Render,
  RenderOptions,
  RenderAPI,
} from '@testing-library/react-native';

const ProviderWrapper: FC = ({ children }) => {
  return <NativeBaseProvider theme={theme}>{children}</NativeBaseProvider>;
};

export const render = (
  component: React.ReactElement<any, string | React.JSXElementConstructor<any>>,
  options?: RenderOptions,
): RenderAPI => {
  return Render(component, { wrapper: ProviderWrapper, ...options });
};
