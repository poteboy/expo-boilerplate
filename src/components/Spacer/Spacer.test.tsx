import React from 'react';
import { Spacer } from './Spacer';
import { NativeBaseProvider } from 'native-base';
import { render } from '@src/util';

describe(`Spacer`, () => {
  it(`renders correctly`, () => {
    const { toJSON } = render(<Spacer size={0} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it(`has one component`, async () => {
    const { queryByTestId, queryAllByTestId } = render(<Spacer size={0} />);
    expect(queryByTestId('spacer')).toBeTruthy();
    expect(queryAllByTestId('spacer')).toHaveLength(1);
  });
});
