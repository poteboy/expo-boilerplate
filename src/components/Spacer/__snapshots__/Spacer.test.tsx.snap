// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacer renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RNCSafeAreaProvider
      onInsetsChange={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        dataSet={Object {}}
        style={
          Array [
            Object {},
            Object {
              "flexShrink": 0,
              "height": 0,
              "width": "auto",
            },
          ]
        }
        testID="spacer"
      />
    </RNCSafeAreaProvider>
  </RNCSafeAreaProvider>
</RNCSafeAreaProvider>
`;
